addons/website_sale_delivery/models/sale_order.py

```diff
@@ -163,3 +163,22 @@ def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, s
                 self._check_carrier_quotation(cr, uid, sale_order, context=context)
 
         return values
+
+    def _get_shipping_country(self, cr, uid, values, context=None):
+        country_ids = set()
+        state_ids = set()
+        values['shipping_countries'] = values['countries']
+        values['shipping_states'] = values['states']
+
+        DeliveryCarrier = self.pool['delivery.carrier']
+        delivery_carrier_ids = DeliveryCarrier.search(cr, SUPERUSER_ID, [('website_published', '=', True)], context=context)
```

Comment:
Hello @mge-odoo ,
Yes, It's Going to restrict the Countries from front-end while getting delivery method from e-commerce, So for Any user we need to check the Country  so it's required. Thanks